pages/configuration/server.conf.rst

```diff
@@ -0,0 +1,407 @@
+    * The maximum size of the HTTP request headers in bytes.
+* ``web_max_initial_line_length = 4096``
+    * The maximal length of the initial HTTP/1.1 line in bytes.
+* ``web_thread_pool_size = 16``
+    * The size of the thread pool used exclusively for serving the web interface.
+
+Elasticsearch
+^^^^^^^^^^^^^
+* ``elasticsearch_hosts = http://node1:9200,http://user:password@node2:19200``
+    * List of Elasticsearch hosts Graylog should connect to.
+    * Need to be specified as a comma-separated list of valid URIs for the http ports of your elasticsearch nodes.
+    * If one or more of your elasticsearch hosts require authentication, include the credentials in each node URI that requires authentication.
+    * Default: ``http://127.0.0.1:9200``
+* ``elasticsearch_connect_timeout = 10s``
+    * Maximum amount of time to wait for successfull connection to Elasticsearch HTTP port.
+    * Default: 10 seconds
+* ``elasticsearch_socket_timeout = 60s``
+    * Maximum amount of time to wait for reading back a response from an Elasticsearch server.
+    * Default: 60 seconds
+* ``elasticsearch_idle_timeout = -1s``
+    * Maximum idle time for an Elasticsearch connection. If this is exceeded, this connection will be tore down.
+    * Default: infinity
+* ``elasticsearch_max_total_connections = 20``
+    * Maximum number of total connections to Elasticsearch.
+    * Default: 20
+* ``elasticsearch_max_total_connections_per_route = 2``
+    * Maximum number of total connections per Elasticsearch route (normally this means per elasticsearch server).
+    * Default: 2
+* ``elasticsearch_max_retries = 2``
+    * Maximum number of times Graylog will retry failed requests to Elasticsearch.
+    * Default: 2
+* ``elasticsearch_discovery_enabled = false``
+    * Enable automatic Elasticsearch node discovery through Nodes Info, see `Elasticsearch Reference » Cluster APIs » Nodes Info <https://www.elastic.co/guide/en/elasticsearch/reference/5.4/cluster-nodes-info.html>`_.
+    * Default: ``false``
+    
+    .. warning:: Automatic node discovery does not work if Elasticsearch requires authentication, e. g. with Shield.
+* ``elasticsearch_discovery_filter = rack:42``
+    * Filter for including/excluding Elasticsearch nodes in discovery according to their custom attributes, see `Elastic Search Reference » Cluster APIs » Node Specification <https://www.elastic.co/guide/en/elasticsearch/reference/5.4/cluster.html#cluster-nodes>`_.
+    * Default: empty
+* ``elasticsearch_discovery_frequency = 30s``
+    * Frequency of the Elasticsearch node discovery.
+    * Default: 30 seconds
+* ``elasticsearch_compression_enabled = false``
+    * Enable payload compression for Elasticsearch requests.
+    * Default: false
+
+Rotation 
+^^^^^^^^
+
+.. attention:: The following settings identified with *!* in this section have been moved to the database in Graylog 2.0. When you upgrade, make sure to set these to your previous 1.x settings so they will be migrated to the database!
```

Comment:
Maybe one more sentence. 
This settings are read once at the very first startup to be the initial settings in the database.